### PR TITLE
Add Hooks example to Themeing > Getting theme without styled components

### PR DESCRIPTION
**Pull request by SpencerDuball**
Adds a sentence and small code example of getting theme outside of styled components when using React Hooks. Right now there is an example using ```withTheme``` for HOCs, but I think this would be a good addition for anyone using Hooks in their codebase to save a little time searching for the way to achieve this in Hooks.

**Before**

before_docs

**After**

after_2

### CODE REVIEW

File: sections/advanced/theming.md [hunk 1]
```diff
@@ -109,6 +109,19 @@ class MyComponent extends React.Component {
 export default withTheme(MyComponent);
 ```
 
+You can also use `useContext` to access the current theme outside of styled components when working with React Hooks.
+
+```jsx
+import { ThemeContext } from 'styled-components';
```
Comment (probablyup) [inline sections/advanced/theming.md]: Can you add the useContext import here too?

Comment (probablyup) [reply sections/advanced/theming.md]: This also requires v4, if you make a subheading for this we have a tagging structure for this (just grep for "v4")

Comment (SpencerDuball) [reply sections/advanced/theming.md]: Sure thing will edit in a minute here.

File: sections/advanced/theming.md [hunk 2]
```diff
@@ -109,6 +111,22 @@ class MyComponent extends React.Component {
...
+You can also use `useContext` to access the current theme outside of styled components when working with React Hooks. | v4
+
+```jsx
+import { useContext } from 'react';
+import { ThemeContext } from 'styled-components';
+
+const MyComponent = () => {
+  const themeContext = useContext(ThemeContext);
+
+  console.log('Current theme: ', this.props.theme);
```
Comment (probablyup) [inline sections/advanced/theming.md]: `console.log('Current theme: ', themeContext);`

File: sections/advanced/theming.md [hunk 3]
```diff
@@ -93,6 +93,8 @@ render(
 
 ### Getting the theme without styled components
 
+#### Using `withTheme` for Higher Order Components
```
Comment (probablyup) [inline sections/advanced/theming.md]: Can you reword this to:

```md
#### via `withTheme` higher-order component
```

File: sections/advanced/theming.md [hunk 4]
```diff
@@ -109,6 +111,22 @@ class MyComponent extends React.Component {
 export default withTheme(MyComponent);
 ```
 
+#### Using React Hooks `useContext` | v4
```
Comment (probablyup) [inline sections/advanced/theming.md]: And this to:

```md
#### via `useContext` React hook | v4
```

Comment (SpencerDuball) [reply sections/advanced/theming.md]: For sure just a moment.

File: sections/advanced/theming.md [hunk 5]
```diff
@@ -109,6 +111,22 @@ class MyComponent extends React.Component {
...
+You can also use `useContext` to access the current theme outside of styled components when working with React Hooks.
+
+```jsx
+import { useContext } from 'react';
+import { ThemeContext } from 'styled-components';
+
+const MyComponent = () => {
+  const themeContext = useContext(ThemeContext);
+
+  console.log('Current theme: ', this.props.theme);
```
Comment (probablyup) [inline sections/advanced/theming.md]: Woops nevermind that approval, this still needs to be fixed. It's `themeContext` not `this.props.theme`.
